In [7]:
-- Visualizando a base de dados
SELECT TOP(10) * FROM baseRH

(10 rows affected)

Total execution time: 00:00:00.019

Nome Gênero Data_Nascimento Idade Nível Escolaridade Área Salário Data_Admissão Data_Demissão Contrato Motivo_Demissão PCD Alimentação Saúde Odont Transporte Creche Combustível Benefícios Mês_admissão Mês_demissão Status Faixa_Idade Tempo_dias Faixa_dias Kylie Rame Masculino 1996-03-17 28 Supervisor Ensino Médio Financeiro 7000 2019-01-01 NULL CLT NULL NULL 1200 1200 1200 1200 NULL 500 5300 1 - NULL 24-34 364 270-364 Dwayne Betun Masculino 1988-02-28 36 Gerente Pós-graduação Recursos Humanos 15000 2019-01-02 NULL CLT NULL NULL 1200 1200 1200 1200 NULL 500 5300 1 - Férias 35-44 363 270-364 Chrisse Oxbrough Masculino 1988-09-22 36 Gerente Ensino Superior Marketing 15000 2019-01-02 NULL CLT NULL NULL 1200 1200 1200 1200 NULL 500 5300 1 - NULL 35-44 363 270-364 Kimmy Fardy Feminino 1988-03-14 36 Gerente Pós-graduação Produção 15000 2019-01-02 NULL CLT NULL NULL 1200 1200 1200 1200 NULL 500 5300 1 - NULL 35-44 363 270-364 Mata Steynor Masculino 1994-06-28 30 Analista Sr Ensino Técnico Produção 5000 2019-01-02 NULL CLT NULL NULL 600 600 600 600 NULL 500 2900 1 - NULL 24-34 363 270-364 Rivi Ofen Feminino 1981-10-19 43 Supervisor Ensino Superior Financeiro 7000 2019-01-03 NULL CLT NULL NULL 1200 1200 1200 1200 NULL 500 5300 1 - NULL 35-44 362 270-364 Ross Tutchell Masculino 1991-05-14 33 Assistente Ensino Médio Produção 2500 2019-01-03 NULL CLT NULL NULL 600 600 600 600 500 NULL 2900 1 - NULL 24-34 362 270-364 Lisa De Antoni Feminino 1961-04-28 63 Estagiário Ensino Médio Administração 1200 2019-01-03 NULL Estágio NULL NULL 200 200 200 200 NULL NULL 800 1 - NULL 55-64 362 270-364 Bethena Josephs Feminino 1978-08-08 46 Analista Jr Ensino Técnico TI 3500 2019-01-03 NULL CLT NULL NULL 600 600 600 600 NULL NULL 2400 1 - NULL 45-54 362 270-364 Kellia Koppel Feminino 1970-07-19 54 Assistente Ensino Superior Recursos Humanos 2500 2019-01-03 NULL CLT NULL NULL 600 600 600 600 NULL NULL 2400 1 - NULL 45-54 362 270-364

In [28]:
-- Total de funcionários por mês de admissão e MoM
WITH CTE AS
(
SELECT
    MONTH(Data_Admissão) AS 'Mês_admissão',
    COUNT(*) AS 'Total'
FROM baseRH
GROUP BY MONTH(Data_Admissão)
)

SELECT 
    Mês_admissão,
    Total,
    FORMAT(1.0 * Total/SUM(Total)OVER(), '0.00%') AS 'Percentual_total',
    FORMAT(1.0 * (Total - LAG(Total,1)OVER(ORDER BY Mês_admissão ASC))/Total, '0.00%') AS 'MoM'
FROM CTE
ORDER BY Mês_admissão ASC

(12 rows affected)

Total execution time: 00:00:00.163

Mês_admissão Total Percentual_total MoM 1 103 10,30% NULL 2 68 6,80% -51,47% 3 87 8,70% 21,84% 4 80 8,00% -8,75% 5 90 9,00% 11,11% 6 91 9,10% 1,10% 7 89 8,90% -2,25% 8 90 9,00% 1,11% 9 72 7,20% -25,00% 10 82 8,20% 12,20% 11 69 6,90% -18,84% 12 79 7,90% 12,66%

O mês com a maior quantidade de contratações foi em Janeiro, no total 103 (10,30%) admitidos.

O mês com o maior crescimento de contratações foi em em Março com 21,84% em relação ao mês anterior.

In [1]:
-- 
SELECT 
    Gênero,
    COUNT(*) AS Total
FROM baseRH
GROUP BY Gênero

(2 rows affected)

Total execution time: 00:00:00.009

Gênero Total Masculino 509 Feminino 491

In [6]:
SELECT
    Área,
    AVG(Salário) AS 'Salario_medio',
    FORMAT(1.0 * AVG(Salário)/SUM(AVG(Salário))OVER(), '0.00%') AS 'Percentual_total'
FROM baseRH
GROUP BY Área
ORDER BY Salario_medio DESC

(9 rows affected)

Total execution time: 00:00:00.387

Área Salario_medio Percentual_total Recursos Humanos 6360 11,21% Vendas 6348 11,19% Marketing 6339 11,17% Financeiro 6326 11,15% Administração 6314 11,13% Operações 6308 11,12% Produção 6281 11,07% TI 6236 10,99% Logística 6236 10,99%

In [32]:
SELECT 
    Idade,
    COUNT(*) AS 'Total'
FROM baseRH
GROUP BY Idade
ORDER BY Idade ASC

(41 rows affected)

Total execution time: 00:00:00.026

Idade Total 24 31 25 22 26 24 27 18 28 19 29 29 30 28 31 25 32 19 33 18 34 25 35 30 36 29 37 23 38 30 39 26 40 24 41 25 42 22 43 25 44 16 45 28 46 26 47 18 48 18 49 30 50 36 51 19 52 24 53 19 54 34 55 29 56 20 57 23 58 20 59 27 60 29 61 26 62 21 63 19 64 26

In [33]:
SELECT 
    Faixa_Idade,
    COUNT(*) AS 'Total'
FROM baseRH
GROUP BY Faixa_Idade
ORDER BY Faixa_Idade ASC

(4 rows affected)

Total execution time: 00:00:00.021

Faixa_Idade Total 24-34 258 35-44 250 45-54 252 55-64 240

In [36]:
SELECT 
    Idade,
    Salário
FROM baseRH

(1000 rows affected)

Total execution time: 00:00:00.015

Idade Salário 28 7000 36 15000 36 15000 36 15000 30 5000 43 7000 33 2500 63 1200 46 3500 54 2500 27 15000 40 5000 44 7000 60 2500 37 15000 31 7000 49 10000 40 3500 33 5000 56 1200 40 10000 31 1200 38 10000 48 1200 60 3500 31 2500 34 2500 32 3500 41 3500 26 15000 31 1200 34 7000 64 2500 52 10000 46 7000 39 2500 63 7000 27 5000 59 3500 45 15000 35 2500 54 2500 50 15000 57 15000 48 15000 57 1200 33 7000 57 2500 49 15000 25 10000 59 5000 54 2500 26 3500 25 2500 26 10000 33 1200 60 10000 51 2500 35 1200 26 1200 57 1200 29 3500 29 10000 53 2500 34 3500 35 1200 28 1200 36 1200 46 10000 46 10000 42 10000 38 7000 51 3500 54 3500 24 1200 50 2500 54 7000 46 2500 62 1200 28 3500 63 7000 46 3500 49 3500 49 15000 56 7000 56 2500 33 3500 54 15000 61 2500 37 3500 27 15000 48 10000 37 5000 42 3500 60 15000 37 15000 36 3500 39 5000 29 3500 48 2500 28 7000 61 2500 51 5000 32 1200 55 15000 40 2500 29 5000 50 10000 37 2500 39 7000 28 5000 59 2500 30 2500 27 15000 57 10000 44 10000 56 5000 39 2500 38 5000 34 2500 53 15000 50 15000 61 3500 52 7000 61 10000 61 15000 38 15000 64 15000 35 5000 45 3500 52 1200 36 15000 40 10000 37 2500 45 2500 28 2500 31 10000 45 10000 43 5000 31 1200 60 1200 24 5000 36 10000 62 1200 43 2500 24 5000 52 2500 47 3500 28 10000 49 7000 54 2500 41 10000 24 10000 61 7000 63 15000 55 7000 47 2500 28 7000 55 3500 53 1200 29 1200 48 7000 56 3500 29 2500 29 3500 62 3500 39 5000 47 7000 34 7000 53 2500 43 5000 63 1200 34 7000 56 3500 52 1200 51 7000 29 2500 27 3500 34 3500 24 5000 52 15000 41 5000 43 3500 58 1200 61 3500 34 7000 61 3500 51 2500 56 7000 47 2500 62 10000 39 3500 50 7000 59 2500 46 10000 39 5000 40 3500 53 2500 60 15000 52 15000 50 15000 54 3500 43 10000 59 3500 27 1200 34 3500 49 5000 61 7000 63 3500 49 3500 36 15000 30 15000 51 10000 59 15000 28 2500 41 7000 52 5000 39 1200 60 7000 46 15000 30 3500 52 3500 47 10000 47 1200 45 3500 58 10000 48 3500 34 3500 26 7000 59 10000 56 15000 42 5000 30 10000 30 2500 40 1200 24 10000 47 15000 55 10000 41 2500 47 5000 30 2500 45 15000 37 10000 49 3500 30 15000 62 2500 35 10000 29 10000 41 10000 50 5000 27 15000 57 10000 49 3500 60 7000 48 5000 35 1200 52 15000 42 5000 49 5000 30 10000 45 2500 36 7000 39 15000 35 2500 50 7000 34 5000 24 2500 50 3500 38 3500 55 15000 50 3500 29 5000 27 3500 54 2500 30 5000 41 10000 57 1200 35 10000 56 2500 44 10000 42 2500 54 7000 58 15000 41 1200 29 10000 49 7000 54 10000 27 3500 38 15000 35 10000 46 2500 43 1200 55 15000 45 2500 37 7000 54 7000 50 15000 38 10000 30 5000 42 5000 45 5000 56 5000 52 10000 35 5000 63 5000 64 1200 45 10000 64 7000 46 15000 42 2500 57 2500 36 1200 62 1200 38 7000 46 15000 50 3500 50 3500 45 10000 48 15000 43 1200 26 5000 33 5000 24 1200 38 15000 40 5000 44 7000 38 10000 31 2500 51 15000 57 5000 46 5000 64 1200 28 15000 39 15000 32 3500 43 5000 30 1200 36 2500 50 10000 32 2500 25 3500 54 2500 59 5000 58 3500 25 5000 56 2500 38 2500 26 10000 61 10000 29 5000 30 5000 56 7000 55 5000 38 10000 44 7000 35 3500 26 3500 50 5000 32 7000 54 10000 59 7000 64 1200 31 15000 31 3500 25 15000 27 7000 49 15000 63 10000 44 15000 40 10000 52 5000 48 3500 53 10000 46 3500 52 5000 33 3500 60 7000 50 2500 26 15000 57 7000 31 10000 30 1200 32 7000 38 5000 31 1200 29 1200 24 2500 51 2500 36 5000 40 2500 35 10000 54 1200 41 7000 49 2500 40 5000 24 15000 52 7000 39 5000 30 10000 55 3500 29 15000 36 5000 64 15000 40 15000 50 1200 49 5000 46 15000 28 5000 24 15000 35 3500 60 1200 59 1200 57 2500 49 5000 35 5000 47 1200 34 7000 64 3500 36 1200 39 5000 60 15000 52 1200 53 15000 41 7000 29 5000 25 7000 40 5000 60 10000 42 7000 53 2500 40 7000 64 2500 42 15000 49 5000 40 3500 62 1200 62 5000 46 10000 36 15000 40 5000 24 1200 52 7000 57 5000 41 15000 64 7000 35 15000 53 3500 62 3500 56 10000 32 1200 26 15000 29 3500 44 10000 54 7000 38 1200 26 7000 25 5000 58 3500 62 1200 59 1200 50 5000 30 1200 60 5000 25 5000 24 2500 48 10000 60 5000 24 7000 38 10000 31 15000 53 3500 43 7000 64 1200 34 1200 64 1200 58 3500 49 5000 39 10000 26 5000 49 2500 64 1000